cilium-sysdump/utils.py

```diff
@@ -165,15 +167,26 @@ def get_pods_status_iterator_by_labels(label_selector, host_ip_filter,
     # feature in the future. The following filter can be extended by
     # modifying the following kubectl custom-columns and the associated
     # grep command.
-    host_ip_filter_cmd = "kubectl get pods --no-headers " \
-        "-o=custom-columns=NAME:.metadata.name," \
-        "HOSTIP:.status.hostIP --all-namespaces | grep -E \"{}\" | " \
-        "awk '{{print $1}}'"
-    host_ip_filter_cmd = host_ip_filter_cmd.format(
-        "|".join(map(str, host_ip_filter)))
+    if node_filter and net.is_ipaddress(node_filter[0]):
```

Comment:
Is it fair to assume that users will only provide node names or IP addresses?
Alternatively, we could apply `is_ip_address` on every element of the original list to turn it into 2 lists, one of node names and another one of IP addresses.

Reply:
Good catch, found a bug if the previous implementation.

Reply:
Summary of the change is:

introduced `get_pods_filtered` which accepts a custom column for `kubectl get pods` and a list of either node names / IPs to filter against the columns. This function allows us to consolidate the duplicate logic to fetch pods for nodes that match an IP or pods that match a node name, thus allowing us to support a mixture of both for `--nodes`.